### ISSUE TITLE
Module bind error if not for training

### ISSUE THREAD

**Issue by lisabug** [dmlc/mxnet#1861]
Module container is flexible to operate whole network such as forward, backward and update.
I want to just use a net for feature extracting, by module.bind(input_shapes=[shapes], for_training=False) but get errors.
Here is a simple script for reproducing error.
```python
import mxnet as mx

data = mx.sym.Variable('data')
fc1 = mx.sym.FullyConnected(data=data, num_hidden=100, name='fc1')
fc2 = mx.sym.FullyConnected(data=fc1, num_hidden=50, name='fc2')

test_mod = mx.mod.Module(fc2)
test_mod.bind(data_shapes=[('data', (5, 100))], for_training=False)
```
error log
```
    test_mod.bind(data_shapes=[('data', (5, 100))], for_training=False)
  File "/home/lu/.virtualenvs/dp/local/lib/python2.7/site-packages/mxnet-0.5.0-py2.7.egg/mxnet/module/module.py", line 233, in bind
    shared_group, logger=self.logger)
  File "/home/lu/.virtualenvs/dp/local/lib/python2.7/site-packages/mxnet-0.5.0-py2.7.egg/mxnet/module/executor_group.py", line 96, in __init__
    self.bind_exec(data_shapes, label_shapes, shared_group)
  File "/home/lu/.virtualenvs/dp/local/lib/python2.7/site-packages/mxnet-0.5.0-py2.7.egg/mxnet/module/executor_group.py", line 141, in bind_exec
    if name in self.param_names]
TypeError: 'NoneType' object has no attribute '__getitem__'
```

**Comment by pluskid** [comment]
Thanks for reporting this! This is a bug. I will post a PR to fix this soon.

**Comment by pluskid** [comment]
This should be fixed after #1843 is merged!

**Comment by pluskid** [comment]
Merged. I'm closing this for now. Feel free to re-open it necessary.
